packages/devtools_app/lib/src/common_widgets.dart

```diff
@@ -793,15 +795,37 @@ class AreaPaneHeader extends StatelessWidget implements PreferredSizeWidget {
               style: theme.textTheme.subtitle2,
               child: title,
             ),
-            ...leftActions,
-            if (rightActions.isNotEmpty) const Spacer(),
-            ...rightActions,
+            ..._buildActions(),
           ],
         ),
       ),
     );
   }
 
+  List<Widget> _buildActions() {
+    return [
+      if (scrollableCenterActions.isEmpty)
+        Expanded(
+          child: Row(
+            children: leftActions,
+          ),
+        ),
+      if (scrollableCenterActions.isNotEmpty) ...[
```

Comment:
More of a Dart clarification question for me - why do `...leftActions` etc. need to be enclosed in a list if that list is spread?

Reply:
Take this scenario:
leftActions = [action1, action2]
scrollableCenterAcitons = [action3, action4]
rightActions = [action5, action6].

Breaking down this method in it's current state with the spread operator, `_buildActions` would return this `List<Widget>`:
```
<Widget>[
  ...[action1, action2,]
  Exanded(ListView(children[action3, action4])),
  ...[action5, action6]
] 
```
which turns into
```
<Widget>[
  action1,
  action2,
  Exanded(child: ListView(children: [action3, action4])),
  action5,
  action6,
] 
```

Without the spread operator on `leftActions`, this is what `_buildActions` would return:
```
<Widget>[
  [action1, action2,] // This is not of type Widget - it is of type List<Widget>, which gives a compilation error
  Exanded(child: ListView(children: [action3, action4])),
  ...[action5, action6]
] 
```